app/channel.go

```diff
@@ -97,7 +97,13 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, shouldBeAdmin
 		_, err = a.Srv().Store.Channel().SaveMember(cm)
 		if histErr := a.Srv().Store.ChannelMemberHistory().LogJoinEvent(user.Id, channel.Id, model.GetMillis()); histErr != nil {
 			mlog.Error("Failed to update ChannelMemberHistory table", mlog.Err(histErr))
-			return histErr
+			var appErr *model.AppError
```

Comment:
I don't see any case where we are actually returning an AppError. So in that case I think we can directly return an AppError without any switch case. Same applies for all of these below.

Reply:
Great, that cleaned a lot.